src/plugins/dashboard/public/application/utils/use/use_dashboard_container.tsx

```diff
@@ -39,30 +55,49 @@ export const useDashboardContainer = (
     const getDashboardContainer = async () => {
       try {
         if (savedDashboardInstance && appState) {
-          let dashboardContainerEmbeddable: DashboardContainer | undefined;
-          try {
-            dashboardContainerEmbeddable = await createDashboardEmbeddable(
-              savedDashboardInstance,
-              services,
-              appState
-            );
-          } catch (error) {
-            console.log(error);
-          }
+          const dashboardContainerEmbeddable = await createDashboardEmbeddable(
+            savedDashboardInstance,
+            services,
+            appState
+          );
+
           setDashboardContainer(dashboardContainerEmbeddable);
         }
       } catch (error) {
-        console.log(error);
+        services.toastNotifications.addWarning({
+          title: i18n.translate('dashboard.createDashboard.failedToLoadErrorMessage', {
+            defaultMessage: 'Failed to load the dashboard',
+          }),
+        });
+        services.history.replace(DashboardConstants.LANDING_PAGE_PATH);
       }
     };
 
     getDashboardContainer();
-  }, [savedDashboardInstance, appState]);
+  }, [savedDashboardInstance, appState, services]);
+
+  useEffect(() => {
+    const incomingEmbeddable = services.embeddable
+      .getStateTransfer(services.scopedHistory())
+      .getIncomingEmbeddablePackage();
+
+    if (
+      incomingEmbeddable &&
+      dashboardContainer &&
+      (!incomingEmbeddable.embeddableId ||
+        !dashboardContainer.getInput().panels[incomingEmbeddable.embeddableId])
+    ) {
+      dashboardContainer.addNewEmbeddable<EmbeddableInput>(
+        incomingEmbeddable.type,
+        incomingEmbeddable.input
+      );
+    }
```

Comment:
```suggestion
    if (
      incomingEmbeddable &&
      dashboardContainer &&
      !dashboardContainer.getInput().panels[incomingEmbeddable.embeddableId]
    ) {
      dashboardContainer.addNewEmbeddable<EmbeddableInput>(
        incomingEmbeddable.type,
        incomingEmbeddable.input
      );
    }
```

Reply:
Or even:
```suggestion
    if (
      incomingEmbeddable &&
      !dashboardContainer.getInput().panels[incomingEmbeddable.embeddableId]
    ) {
      dashboardContainer?.addNewEmbeddable<EmbeddableInput>(
        incomingEmbeddable.type,
        incomingEmbeddable.input
      );
    }
```

Reply:
Got rid of unnecessary conditional statements: 

```
    if (
      incomingEmbeddable &&
      !dashboardContainer?.getInput().panels[incomingEmbeddable.embeddableId!]
    ) {
      dashboardContainer?.addNewEmbeddable<EmbeddableInput>(
        incomingEmbeddable.type,
        incomingEmbeddable.input
      );
    }

```